app.py

```diff
@@ -0,0 +1,313 @@
+            "dateTime": f"{datetime.utcnow().isoformat()}Z",
+            "transactionId": str(uuid.uuid4())
+        },
+        "payload": {
+            "refusal": {
+                "type": "HARD_REFUSAL",
+                "report": "respondent unwilling",
+                "agentId": "110001",
+                "collectionCase": {
+                    "id": random_case['case_id']
+                },
+                "contact": {
+                    "title": "Ms",
+                    "forename": "jo",
+                    "surname": "smith",
+                    "telNo": "+447890000000"
+                },
+                "address": {
+                    "addressLine1": "1a main street",
+                    "addressLine2": "upper upperingham",
+                    "addressLine3": "",
+                    "townName": "upton",
+                    "postcode": "UP103UP",
+                    "region": "E",
+                    "uprn": "123456789"
+                }
+            }
+        }
+    }
+
+    message = {
+        "type": "RABBIT",
+        "topic": "event.respondent.refusal",
+        "delay": random_delay,
+        "message_body": json.dumps(message_contents)
+    }
+
+    messages_to_send.append(message)
+
+
+def prepare_fulfilment_requested(random_delay, random_case):
+    message_contents = {
+        "event": {
+            "type": "FULFILMENT_REQUESTED",
+            "source": "CONTACT_CENTRE_API",
+            "channel": "CC",
+            "dateTime": f"{datetime.utcnow().isoformat()}Z",
+            "transactionId": str(uuid.uuid4())
+        },
+        "payload": {
+            "fulfilmentRequest": {
+                "fulfilmentCode": "UACHHT1",
+                "caseId": random_case['case_id'],
+                "address": {},
+                "contact": {
+                    "telNo": "+447876224123"
+                }
+            }
+        }
+    }
+
+    message = {
+        "type": "RABBIT",
+        "topic": "event.fulfilment.request",
+        "delay": random_delay,
+        "message_body": json.dumps(message_contents)
+    }
+
+    messages_to_send.append(message)
+
+
+def prepare_undelivered_mail_reported(random_delay, random_case):
+    message_contents = {
+        "transactionId": str(uuid.uuid4()),
+        "dateTime": f"{datetime.now().replace(microsecond=0).isoformat()}",
+        "questionnaireId": random_case['qid'],
+        "unreceipt": False
+    }
+
+    message = {
+        "type": "PUBSUB",
+        "topic": Config.QM_UNDELIVERED_TOPIC_NAME,
+        "project": Config.QM_UNDELIVERED_PROJECT_ID,
+        "delay": random_delay,
+        "message_body": json.dumps(message_contents)
+    }
+
+    messages_to_send.append(message)
+
+
+def prepare_messages_to_be_sent():
+    for _ in range(total_messages_to_send):
+        random_message_type = message_type_randomiser[random.randint(0, len(message_type_randomiser) - 1)]
+        random_delay = float(random.randint(0, 1000 / message_rate)) / 1000.0
```

Comment:
This can be made safer with something like:
        top_range = int(1000/message_rate) + 1
        random_delay = float(random.randint(0, top_range)) / 1000.0

Other wise it needs to be a factor of 1000 to stop it getting a none int error.  Or there might be a randFloat option?

Reply:
If the random delay is in seconds, shouldn't this be something like
```python
random_delay = random.random() / message_rate
```
(`random.random()` return a random float in [0, 1))

Reply:
that sounds easier